src/server_manager/web_app/app.ts

```diff
@@ -375,7 +375,7 @@ export class App {
 
   private async loadManualServers() {
     for (const server of await this.manualServerRepository.listServers()) {
-      this.addServer(server);
+      await this.addServer(server);
```

Comment:
addServer is a synchronous method. There's no point in calling await.
When addServer returns, the server will already have been added to the list.

Reply:
Thanks for helping figure out the root of the issue.

For posterity, the issue was that the Polymer `dom-repeat` was rendering newly added server views asynchronously. This lead to a race condition where the server view was not attached to the DOM before attempting to show the server.